### PR TITLE
TRUNK-5818 Enable module devs to write Junit 4 & 5 tests

### PR DESCRIPTION
**Pull request by achilep**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -469,6 +469,30 @@
...
+			<dependency>
+				<groupId>org.junit.vintage</groupId>
+				<artifactId>junit-vintage-engine</artifactId>
+				<version>${junitVersion}</version>
+				<scope>test</scope>
+			</dependency>
+			<dependency>
+				<groupId>org.mockito</groupId>
+				<artifactId>mockito-junit-jupiter</artifactId>
+				<version>3.4.6</version>
```
Comment (teleivo) [inline pom.xml:493]: can you also please extract the mockitoVersion and reuse it for mockito core and its junit-jupiter library

File: api/pom.xml [hunk 2]
```diff
@@ -287,26 +287,18 @@
 	   <dependency>
 		   <groupId>org.junit.jupiter</groupId>
 		   <artifactId>junit-jupiter-api</artifactId>
-		   <version>${junitVersion}</version>
```
Comment (teleivo) [inline api/pom.xml:290]: can you please check if you now do a `mvn dependency tree` are these libs still with scope test?

**Comment by achilep** [comment]
test
Screenshot from 2020-08-06 15-43-40

api
Screenshot from 2020-08-06 15-45-33
Screenshot from 2020-08-06 15-45-51
web:
Screenshot from 2020-08-06 15-47-38

File: test/pom.xml [hunk 3]
```diff
@@ -34,6 +34,22 @@
...
+	   </dependency>
+	   <dependency>
+		   <groupId>org.junit.jupiter</groupId>
+		   <artifactId>junit-jupiter-engine</artifactId>
+	   </dependency>
+	   <dependency>
+		   <groupId>org.junit.vintage</groupId>
+		   <artifactId>junit-vintage-engine</artifactId>
+	   </dependency>
+	   <dependency>
```
Comment (teleivo) [inline test/pom.xml:49]: please group the mockito-junit with the mockito core dependency. make sure that is the case in all pom.xml . sorry that I did not see this earlier

**Pull request merged**